lib/rules/keyframe-block-no-duplicate-selectors/index.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+
+const ruleName = 'keyframe-block-no-duplicate-selectors';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (selector) => `Unexpected duplicate "${selector}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/keyframe-block-no-duplicate-selectors',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkAtRules(/^(-(moz|webkit)-)?keyframes$/i, (atRuleKeyframes) => {
+			const selectors = new Set();
+
+			atRuleKeyframes.walkRules((keyframeRule) => {
+				const ruleSelectors = keyframeRule.selectors;
+
```

Comment:
Let's make sure the selector is standard syntax here using https://github.com/stylelint/stylelint/blob/main/lib/utils/isStandardSyntaxSelector.js

Reply:
Add this in the `.forEach` - is this what you were thinking?